src/core/mac/mac.cpp

```diff
@@ -1262,6 +1262,18 @@ void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError
 
     case kOperationTransmitDataDirect:
         mCounters.mTxData++;
+#if OPENTHREAD_CONFIG_MAC_RETRY_SUCCESS_HISTOGRAM_ENABLE
+        if (aError != OT_ERROR_NONE)
+        {
+            mCounters.mTxDirectMaxRetryExpiry++;
+        }
+        else if ((0 < mSubMac.GetTransmitRetries()) &&
```

Comment:
Let's include also the situation where the packet was sent in the first try (without retries).